### PR TITLE
[t3]refactored controllers, added service

### PR DESCRIPTION
**Pull request by rl3250**
qq: i think it's still not clear where the deposit/withdraw lives in terms of resources relative to users.

also i left the commented code just for cr.

### CODE REVIEW

File: src/main/java/com/lgtm/easymoney/controllers/UserController.java [hunk 1]
```diff
@@ -1,67 +1,93 @@
...
 
-        var userCreated = userRepository.save(user);
+        return success ? ResponseEntity.status(HttpStatus.OK).body(res) :
+                ResponseEntity.status(HttpStatus.NOT_FOUND).body(new ErrorRsp(new ArrayList<>(Arrays.asList("i think we need to refactor this")), "Error in making a withdraw."));
 
-        return ResponseEntity.status(HttpStatus.CREATED).body(new RegisterRsp(userCreated.getId()));
     }
 
-    // TODO login, get profile, update profile
+//    @PutMapping("/update")
```
Comment (rl3250) [inline src/main/java/com/lgtm/easymoney/controllers/UserController.java:58]: to reviewers: will remove this once approved. intentionally left those here for reference

File: src/main/java/com/lgtm/easymoney/controllers/UserController.java [hunk 2]
```diff
@@ -1,67 +1,93 @@
...
+    @PutMapping("/withdraw")
+    public ResponseEntity<?> withdraw(@Valid @RequestBody BalanceReq req) {
+        // get params
+        Long uid = req.getUid();
+        BigDecimal amount = req.getAmount();
+        // make a withdraw
+        User user = userService.getUserByID(uid);
+        boolean success = userService.makeAWithdraw(user, amount);
+        // payload
+        BalanceRsp res = new BalanceRsp();
```
Comment (rl3250) [inline src/main/java/com/lgtm/easymoney/controllers/UserController.java:49]: qq: should we have a service for handling responses?

Comment (howieraem) [reply src/main/java/com/lgtm/easymoney/controllers/UserController.java:49]: I think you can move the payload initialization to service, example: https://github.com/osopromadze/Spring-Boot-Blog-REST-API/blob/master/src/main/java/com/sopromadze/blogapi/service/TagService.java

**Review by howieraem — APPROVED** [review]
lg

**Review by KenXiong123 — APPROVED** [review]

**Pull request merged**